### PR TITLE
Feature/articles

### PR DESCRIPTION
**Pull request by Anastasiia-Ihn**

### CODE REVIEW

File: controllers/v2/ArticleController.ts [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+
+    const data = await Article.find({
+      $or: [{ title: searchQuery }],
+    })
+      .skip(skip)
+      .limit(itemsPerPage);
+
+    if (!data.length) {
+      return res.status(404).json({ message: "Articles not found" });
+    }
```
Comment (Simkav) [inline controllers/v2/ArticleController.ts]: move [here](https://github.com/baza-trainee/baza_trainee_back/pull/5/files#diff-ede08c51bb56d2ee50e45fcc14637e230b79df92bdcceb608761c2316449c6d6R40)

condition based on totalDocuments

File: controllers/v2/ArticleController.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  }
+};
+
+export const getArticles = async (req: Request, res: Response) => {
+  const { search, page, limit } = req.query;
+
+  const searchQuery = new RegExp(search as string, "i");
+
+  const currentPage = parseInt(page as string) || 1;
+  const itemsPerPage = parseInt(limit as string) || 9;
```
Comment (Simkav) [inline controllers/v2/ArticleController.ts:33]: must be limit like 25

File: controllers/v2/ArticleController.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+    });
+  }
+};
+
+export const updateArticle = async (req: Request, res: Response) => {
+  try {
+    const id = req.params.id;
+    const newDataArticle = req.file?.filename
+      ? { ...req.body, imageUrl: req.file?.filename }
+      : req.body;
```
Comment (Simkav) [inline controllers/v2/ArticleController.ts]: const newDataArticle = req.body
 if (req.file?.filename) newDataArticle.imageUrl = req.file?.filename

File: controllers/v2/ArticleController.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+    const oldArticle = await Article.findById(id);
+
+    if (!oldArticle) {
+      return res.status(404).json({ message: "Article not found" });
+    }
```
Comment (Simkav) [inline controllers/v2/ArticleController.ts:97]: do at start

File: models/v2/BlogArticle.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+import { Schema, model } from "mongoose";
+import { IBlogArticle } from "../../types";
+
+const BlogArticleSchema = new Schema<IBlogArticle>({
+  title: { type: String, required: true },
+  text: { type: String, required: true },
+  date: Number,
+  imageUrl: { type: String, required: true },
+});
```
Comment (Simkav) [inline models/v2/BlogArticle.ts]: date:Number?

File: settings.ts [hunk 6]
```diff
@@ -1,21 +1,22 @@
...
+  allowCreateDocumentWithoutFile: false, //If true, the document can be created with the URL.
+  allowUserRegistration: true,
+  maxNumberOfItems: {
+    heroSlider: 5,
+    testimonialsSlider: 23,
+  },
+  fileUploadMethod: "disk", //disk or s3 storage
+  fileUploadFolderName: "baza-static",
+  textToTranslateTextMaxLength: 700,
+  tokenExpiration: "30d",
```
Comment (Simkav) [inline settings.ts:21]: why

Comment (Anastasiia-Ihn) [reply settings.ts:21]: it is Prettier 
I added only 10 line

File: models/v2/BlogArticle.ts (hunk 5 above)
Comment (Anastasiia-Ihn) [reply models/v2/BlogArticle.ts]: This type was chosen in other similar old schemas, and I am writing according to these schemas.

**Pull request merged**